hp_list/main.c

```diff
@@ -14,6 +14,110 @@
 #include <string.h>
 #include <threads.h>
 
+#ifdef RUNTIME_STAT
+/*
+ * "rtry" the number of retries in the __list_find function.
+ * "cons" the number of wait-free contains in the __list_find function that curr
+ * pointer pointed.
+ * "trav" the number of list element traversal in the __list_find function.
+ * "fail" the number of CAS() failures.
+ * "del" the number of list_delete operation failed and restart again.
+ * "ins" the number of list_insert operation failed and restart again.
+ * "load"  is the number of atomic_load operation in list_delete, list_insert
+ * and __list_find.
+ * "store" is the number of atomic_store operation in list_delete, list_insert
+ * and __list_find.
+ */
+static atomic_uint_fast64_t rtry = 0, cons = 0, trav = 0, fail = 0;
+static atomic_uint_fast64_t del = 0, ins = 0;
+static atomic_uint_fast64_t load = 0, store = 0;
+static atomic_uint_fast64_t deletes = 0, inserts = 0;
+
+enum {
+    TRACE_NOTHING = 0,
+    TRACE_TRIES,
+    TRACE_WAIT_FREE_CONS,
+    TRACE_TRAVERSAL,
+    TRACE_DEL,
+    TRACE_INS,
+    TRACE_INSERTS,
+    TRACE_DELETES
+};
+
+#define CAS(obj, expected, desired)                                          \
+    ({                                                                       \
+        bool __ret = atomic_compare_exchange_strong(obj, expected, desired); \
+        if (!__ret)                                                          \
+            atomic_fetch_add(&fail, 1);                                      \
+        __ret;                                                               \
+    })
+#define ATOMIC_LOAD(obj)            \
+    ({                              \
+        atomic_fetch_add(&load, 1); \
+        atomic_load(obj);           \
+    })
+#define ATOMIC_STORE_EXPLICIT(obj, desired, order)  \
+    do {                                            \
+        atomic_fetch_add(&store, 1);                \
+        atomic_store_explicit(obj, desired, order); \
+    } while (0)
+#define TRACE_ATOMIC_FECTH_ADD(obj, arg, ops) \
+    ({                                        \
+        if (ops)                              \
```

Comment:
You can make use of `##`, so that `ops` would be shortened. That is,
```c
TRACE_ATOMIC_FECTH_ADD(obj, arg, DEL); // No TRACE_ prefix in the last parameter
```